plugins/modules/managed_cluster_addon.py

```diff
@@ -0,0 +1,190 @@
+          an addon will be enabled. If set to C(absent), an existing addon will be disabled.
+        type: str
+        default: present
+        choices: [ absent, present ]
+        required: False
+'''
+
+EXAMPLES = r'''
+- name: "Enabled cluster-proxy addon"
+  ocmplus.cm.managed_cluster_addon:
+    state: present
+    hub_kubeconfig: /path/to/hub/kubeconfig
+    managed_cluster: example-cluster
+    addon_name: cluster-proxy
+    wait: True
+    timeout: 120
+
+- name: "Disabled cluster-proxy addon"
+  ocmplus.cm.managed_cluster_addon:
+    state: absent
+    hub_kubeconfig: /path/to/hub/kubeconfig
+    managed_cluster: example-cluster
+    addon_name: cluster-proxy
+    wait: True
+    timeout: 60
+'''
+
+RETURN = r'''
+result:
+    description: message describing the addon enabled/disabled successfully done.
+    returned: success
+    type: str
+err:
+  description: Error message
+  returned: when there's an error
+  type: str
+  sample: null
+'''
+
+import traceback
+
+from ansible.module_utils.basic import AnsibleModule, env_fallback, missing_required_lib
+from ansible_collections.ocmplus.cm.plugins.module_utils.import_utils import get_managed_cluster
+from ansible_collections.ocmplus.cm.plugins.module_utils.managed_cluster_addons import (
+    application_manager,
+    cert_policy_controller,
+    cluster_proxy,
+    iam_policy_controller,
+    managed_serviceaccount,
+    policy_controller,
+    search_collector,
+)
+from pkgutil import iter_modules
+from pathlib import Path
+import os
+
+IMP_ERR = {}
+try:
+    import kubernetes
+    from kubernetes.dynamic.exceptions import NotFoundError
+except ImportError as e:
+    IMP_ERR['k8s'] = {'error': traceback.format_exc(),
+                      'exception': e}
+
+
+def execute_module(module: AnsibleModule):
+    if 'k8s' in IMP_ERR:
+        # we will need k8s for this module
+        module.fail_json(msg=missing_required_lib('kubernetes'),
+                         exception=IMP_ERR['k8s']['exception'])
+
+    addon_name = module.params['addon_name']
+    managed_cluster_name = module.params['managed_cluster']
+    hub_kubeconfig = kubernetes.config.load_kube_config(
+        config_file=module.params['hub_kubeconfig'])
+    hub_client = kubernetes.dynamic.DynamicClient(
+        kubernetes.client.api_client.ApiClient(configuration=hub_kubeconfig)
+    )
+    wait = module.params['wait']
+    timeout = module.params['timeout']
+    if timeout is None or timeout <= 0:
+        timeout = 60
+
+    state = module.params['state']
+    if state == 'present':
+        managed_cluster = get_managed_cluster(hub_client, managed_cluster_name)
+        if managed_cluster is None:
+            module.fail_json(
+                msg=f'failed to get managedcluster {managed_cluster_name}')
+
+    enabled = True if state == 'present' else False
+    new_addon_name = addon_name.replace('-', '_')
+    new_addon = globals()[new_addon_name](module, enabled, hub_client,
```

Comment:
oh, so the addon classes are used here? this looks pretty hard to understand, and will it cause any type None or nil pointer issues?

Reply:
if import works, there is no problem